### DISCUSSION TITLE
QUESTION: MAUI Android not able to load .mbtiles ❓

### DISCUSSION THREAD

**Discussion by alvaroalfaro612** [Mapsui/Mapsui#1854] [Q&A]
Hello,

I am not able to load .mbtiles for  offine maps with Mapsui in **Android**, using **MAUI .NET**.

I have added the .mbtiles to the project as MauiAsset and also as embedded resource, but always I get **SQLite.SQLiteException: 'Could not open database file: countries-raster.mbtiles (CannotOpen)'**.

I am using the same code I have seen in other examples, by using **BruTile** :

```
        var pathSql = new SQLiteConnectionString("countries-raster.mbtiles", true);
        var layer = new TileLayer(new MbTilesTileSource(pathSql, null, MbTilesType.None, true, true));
        mapControl.Map?.Layers.Add(layer);
```

I have tried several combinations for the path and the way of loading the .mbtile, but none of them are working for me...

**Comment by charlenni** [comment]
Perhaps something is wrong with your SQLite file. Did you try to open it with a SQLite Manager to view the content?

**Comment by pauldendulk** [comment]
We use embedded resources in our samples because it is easier that way to support all the different platforms. These embedded resources are deployed as file in the static constructor of the samples. You only need the file itself, not the embedded resource. Perhaps first start with a File.Open (without Mapsui or MbTiles) to see if you can access the file at all.

**Comment by alvaroalfaro612** [comment]
Thanks for your help.

I have open the mbtile with SQLite viewer and seems to be correct. I think the problem is that I am not able to get the correct path to the file...

I have used embedded resources, and with these lines of code I get the list of embedded resources, and my mbtiles is there:

```
        var executingAssembly = Assembly.GetExecutingAssembly();
        string folderName = string.Format("{0}.Resources.Folder", executingAssembly.GetName().Name);
        List<string> embeddedResourcesList = executingAssembly.GetManifestResourceNames();
```

The problem is that I need a string with the **path** of the file to provide to the SQLite connection:

`var pathSQL = new SQLiteConnectionString(path, true);`

@pauldendulk , may you provide the link with  the sample you mentioned, using offline maps with mbtiles?

Thanks in advance.
